libiocage/lib/Release.py

```diff
@@ -310,75 +369,129 @@ def update(self):
         dataset.snapshot(snapshot_name, recursive=True)
 
         jail = libiocage.lib.Jail.Jail({
-            "uuid"              : str(uuid.uuid4()),
-            "basejail"          : False,
-            "allow_mount_nullfs": "1",
-            "release"           : self.name,
-            "securelevel"       : "0"
-        },
+                "uuid"              : str(uuid.uuid4()),
+                "basejail"          : False,
+                "allow_mount_nullfs": "1",
+                "release"           : self.name,
+                "securelevel"       : "0"
+            },
+            new=True,
             logger=self.logger,
             zfs=self.zfs,
             host=self.host
```

Comment:
None of these are actually defined from what I can see.

Reply:
look out for `libiocage.lib.helpers.init_`. That's where they are initialized.

Reply:
Then in `__init__` we should have something similar to `self.logger, self.zfs, self.host = libiocage.lib.helpers.init_`, as that's not only non-obvious, it requires chasing context around when debugging things.

Reply:
If it improves the readability we could slightly change the interface of our init_ helpers to return the existing or new initialized instance and manually set it, e.g.

```python
self.logger = libiocage.lib.helpers.init_logger(logger)
```

instead of

```python
libiocage.lib.helpers.init_logger(self, logger)
```

We should tackle it in a general refactoring iteration. Do you want me to create a ticket for that?